lesson04.md

```diff
@@ -0,0 +1,464 @@
+<!-- .slide: id="lesson4" -->
+
+# Basic Frontend - Spring 2021
+
+Lesson 4, Thursday, 2021-03-11
+
+---
+
+### Recap: data types
+
+Number:
+```js
+42
+-3.1415
+```
+
+String:
+```js
+"John"
+'Berlin'
+```
+
+Boolean:
+```js
+true
+false
+```
+
+Undefined:
+```js
+undefined
+```
+
+---
+
+### Recap operators
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * / **` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `30 !== 25`<br/>`20 >= 18` |
+
+---
+
+### What does it print?
+
+```js
+let name = "Sevtap";
+console.log("Hello " + name);
+```
+
+Answer: It prints "Hello Sevtap" <!-- .element: class="fragment" -->
+
+---
+
+### What does it print?
+
+```js
+let greeting = "Hello";
+let name = "Sevtap";
+console.log(greeting + " " + firstName);
+```
+
+Answer: Error, "firstName" is not defined <!-- .element: class="fragment" -->
+
+---
+
+### What does it print?
+
+```js
+console.log("It is " + temperature + " degrees");
+let temperature = 22;
+```
+
+Answer: Error, "temperature" is not defined <!-- .element: class="fragment" -->
+
+---
+
+## Pass-by-value
+
+---
+
+## Passing basic data types
+
+* What do you think `x` and `y` are at the end of the program?
+
+```js
+let x = 42;
+let y = x;
+x = 43;
+console.log(x);
+console.log(y);
+```
+
+Answer: x is 43, y is 42 <!-- .element: class="fragment" -->
+
+---
+
+### Pass-by-value
+
+* All basic data types (number, string, boolean) are passed by value.
+* If you assign a variable to another variable, its value is _copied_.
+
+---
+
+## Combined Operators
+
+---
+
+### Operators
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * /` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `temperature !== 25`<br/>`age >= 18` |
+| **Combined Operators** | `+= -= *= /= ++ --` | `a *= 2`<br />`count++` |
+
+---
+
+### Combined Operators
+
+```js
```

Comment:
Let's emphasise that term appearing on the left side of the = sign must be a variable. So for example 3 += x is wrong. Because that means 3 = 3 + x; We can not change the value of number 3 that is why. 
When reading code which involves = sign, always read it like this, the value on the right side is assigned to/ put into the variable in the left side.